### PR TITLE
added IO to broadcastable

### LINKED ISSUE #42063: IO objects should be broadcast "scalars"
This doesn't seem right:
```jl
julia> println.(stdout, [1,2,3])
ERROR: MethodError: no method matching length(::Base.TTY)
Closest candidates are:
  length(::Union{Base.KeySet, Base.ValueIterator}) at abstractdict.jl:58
  length(::Union{LinearAlgebra.Adjoint{T, S}, LinearAlgebra.Transpose{T, S}} where {T, S}) at /Users/julia/buildbot/worker/package_macos64/build/usr/share/julia/stdlib/v1.6/LinearAlgebra/src/adjtrans.jl:195
  length(::Base.Generator) at generator.jl:50
  ...
Stacktrace:
 [1] _similar_for(c::UnitRange{Int64}, #unused#::Type{Any}, itr::Base.TTY, #unused#::Base.HasLength)
   @ Base ./array.jl:575
 [2] _collect(cont::UnitRange{Int64}, itr::Base.TTY, #unused#::Base.HasEltype, isz::Base.HasLength)
   @ Base ./array.jl:608
 [3] collect(itr::Base.TTY)
   @ Base ./array.jl:602
 [4] broadcastable(x::Base.TTY)
   @ Base.Broadcast ./broadcast.jl:682
 [5] broadcasted(::Function, ::Base.TTY, ::Vector{Int64})
   @ Base.Broadcast ./broadcast.jl:1312
 [6] top-level scope
   @ REPL[9]:1
```
Should be an easy fix — just add `IO` (the abstract supertype of all I/O streams) to the `Union` on [this line](https://github.com/JuliaLang/julia/blob/92337b560a3a5b71181ed420bea1890d196517a6/base/broadcast.jl#L700), and add a test.

### PR DESCRIPTION
**Pull request by ArunS-tack**
closes #42063.

### CODE REVIEW

File: test/broadcast.jl [hunk 1]
```diff
@@ -1056,3 +1056,8 @@ end
     @test Broadcast.BroadcastFunction(+)(2:3, 2:3) == 4:2:6
     @test Broadcast.BroadcastFunction(+)(2:3, 2:3) isa AbstractRange
 end
+
+@testset "#42063" begin
+    A = println.(stdout, [1,2,3,4])
```
Comment (sostock) [inline test/broadcast.jl:1061]: Printing to `stdout` will just add noise to the output when testing. You could print to an `IOBuffer` instead and check that the resulting content is correct.

Comment (ArunSanganal) [reply test/broadcast.jl]: I guess IOBuffer works even without this pull request.

Comment (sostock) [reply test/broadcast.jl:1061]: Does `IOBuffer` work for you without this PR? For me it doesn’t:
```
julia> println.(IOBuffer(), [1,2,3])
ERROR: MethodError: no method matching length(::IOBuffer)
```

Comment (sostock) [reply test/broadcast.jl:1061]: Ah, I see. You tried to print to the *type* `IOBuffer`, not to an instance of an `IOBuffer`.

Comment (ArunSanganal) [reply test/broadcast.jl:1061]: my bad, I forgot to add those parentheses. You are right, it does not.

Comment (sostock) [reply test/broadcast.jl:1061]: I meant something like this:
```julia
io = IOBuffer()
println.(io, [1,2,3])
@test String(take!(io)) == "1\n2\n3\n"
```

File: test/broadcast.jl [hunk 2]
```diff
@@ -1056,3 +1056,8 @@ end
     @test Broadcast.BroadcastFunction(+)(2:3, 2:3) == 4:2:6
     @test Broadcast.BroadcastFunction(+)(2:3, 2:3) isa AbstractRange
 end
+
+@testset "#42063" begin
+    A = println.(IOBuffer(), [1,2,3,4])
+    @test A isa Vector{Nothing}
```
Comment (stevengj) [inline test/broadcast.jl:1062]: I would do
```jl
buf = IOBuffer()
@test println.(buf, [1,2,3]) == [nothing, nothing, nothing]
@test String(take!(buf)) == "1\n2\n3\n"
```
to make sure that the buffer has the expected contents.

**Comment by vchuravy** [comment]
Please rebase and squash onto latest master

**Review by stevengj — APPROVED** [review]

**Comment by stevengj** [comment]
Rebased.

**Comment by stevengj** [comment]
Should be good to merge?

**Pull request merged**